controller/kv/helper.go

```diff
@@ -1582,6 +1616,135 @@ func (m clusterHelper) GetScannerDB(store string) []*share.CLUSScannerDB {
 	return dbs
 }
 
+func (m clusterHelper) GetAvailableScanners() []*share.CLUSScanner {
+	scanners := make([]*share.CLUSScanner, 0)
+	if keys, err := cluster.GetStoreKeys(share.CLUSScannerStore); err == nil {
+		for _, key := range keys {
+			var scanner share.CLUSScanner
+			value, _, err := m.get(key)
+			if err != nil {
+				log.WithFields(log.Fields{"key": key, "error": err}).Error("Failed to get scanner")
+				continue
+			}
+			if value != nil {
+				if err := nvJsonUnmarshal(key, value, &scanner); err != nil {
+					log.WithFields(log.Fields{"key": key, "error": err}).Error("Failed to unmarshal scanner")
+					continue
+				}
+				if scanner.ID != share.CLUSScannerDBVersionID {
+					scanners = append(scanners, &scanner)
+				}
+			}
+		}
+	}
+
+	return scanners
+}
+
+func (m clusterHelper) PickLeastLoadedScanner(ScannerHealthCheckFunc func(string, time.Duration) error) (*share.CLUSScanner, error) {
+	scanners := m.GetAvailableScanners()
+	if len(scanners) == 0 {
+		return nil, share.ErrNoScannerFound
+	}
+
+	// To minimize CAS conflicts and improve fairness, sort scanners by ScanCredit (descending).
+	// This ensures we try to consume credits from the most available scanner first.
+	sort.Slice(scanners, func(i, j int) bool {
+		return scanners[i].ScanCredit > scanners[j].ScanCredit
+	})
+
+	for _, scanner := range scanners {
+		if scanner == nil {
+			log.Error("scanner is nil, skipping")
+			continue
+		}
+
+		if ScannerHealthCheckFunc != nil {
+			if err := ScannerHealthCheckFunc(scanner.ID, time.Second*10); err != nil {
+				log.WithFields(log.Fields{"scanner": scanner.ID, "error": err}).Error("Failed to ping scanner")
+				continue
+			}
+		}
```

Comment:
Question: Do we really need this?  My understanding is that we've had logics to ping these scanners [here](https://github.com/neuvector/neuvector/blob/52396fc8f4dca37fbc3665d458e412b4d5a29c51/controller/cache/cache.go#L1759).  Adding the health check here might help, but there is still a window that you assign a job to a scanner that repeatedly restarts.